pandas/tests/extension/base/ops.py

```diff
@@ -27,13 +29,23 @@ def _get_expected_exception(
         # The self.obj_bar_exc pattern isn't great in part because it can depend
         #  on op_name or dtypes, but we use it here for backward-compatibility.
         if op_name in ["__divmod__", "__rdivmod__"]:
-            return self.divmod_exc
-        if isinstance(obj, pd.Series) and isinstance(other, pd.Series):
-            return self.series_array_exc
+            result = self.divmod_exc
+        elif isinstance(obj, pd.Series) and isinstance(other, pd.Series):
+            result = self.series_array_exc
         elif isinstance(obj, pd.Series):
-            return self.series_scalar_exc
+            result = self.series_scalar_exc
         else:
-            return self.frame_scalar_exc
+            result = self.frame_scalar_exc
+
+        if using_pyarrow_string_dtype() and result is not None:
```

Comment:
I am open to other suggestions if we keep in mind that this is a temporary solution until we enable the option by default which will make testing this a lot easier

Reply:
this is pretty ugly but since its temporary (and im way late in reviewing) im not inclined to worry about it.  There are a bunch of similar things that i think could be improved by catching pa.lib.ArrowNotImplementedError and re-raising as TypeError for cases that are not _supposed_ to work